include/boost/url/impl/const_string.hpp

```diff
@@ -103,35 +82,31 @@ const_string(
     Allocator const& a)
 {
     char* dest;
-    p_ = construct(
-        s.size(), a, dest);
-    std::memcpy(dest,
-        s.data(), s.size());
+    *this = const_string(s.size(), a, dest);
+    std::memcpy(dest, s.data(), s.size());
 }
 
 const_string::
 const_string(
     const_string const& other) noexcept
     : string_view(other)
-    , p_(other.p_)
+    , refs_(other.refs_)
+    , alloc_(other.alloc_)
 {
-    if(p_)
-        ++p_->refs;
+    increment_refs();
 }
 
 const_string&
 const_string::
 operator=(
     const_string const& other) & noexcept
 {
-    if( p_ &&
-        --p_->refs == 0)
-        p_->destroy();
-    p_ = other.p_;
-    if(p_)
-        ++p_->refs;
-    static_cast<string_view&>(
-        *this) = other;
+    decrement_refs();
+    refs_ = other.refs_;
+    alloc_ = other.alloc_;
+    increment_refs();
+    static_cast<string_view&>(*this) =
+        static_cast<const string_view&>(other);
```

Comment:
the second static_cast is probably not needed

Reply:
Yes. It does work without the static_cast here, but it emits a warning about slicing.